baseservice/iodevices/base/input_transaction.py

```diff
@@ -96,24 +98,32 @@ class InputTransactionScope(InputTransaction):
     a helper class for reading several messages inside a transaction scope.
     """
 
-    def __init__(self, device: Optional['InputDevice'] = None, with_transaction: bool = True):
+    def __init__(self, device: 'InputDevice', with_transaction: bool = True):
         """
-
         :param device: the input device to read the messages from
         :param with_transaction: 'True' if we should actually use transactions, 'False' otherwise
         """
         super().__init__(device)
         self._with_transaction = with_transaction
         self._transactions: List[InputTransaction] = []
 
-    def read_message(self, timeout: Optional[float] = 0) -> Union[Tuple[Message, DeviceHeaders], Tuple[None, None]]:
-        message, device_headers, transaction = self.device.read_message(timeout=timeout,
-                                                                        with_transaction=self._with_transaction)
+    def read_message(self, timeout: Optional[float] = None) -> Optional['ReadResult']:
+        """
+        this method returns a message from the device, and adds its transaction to the transaction scope
+
+        :param timeout: an optional timeout (in seconds) to wait for the device to return a message.
```

Comment:
I think it would be more readable if you explcitly explain that if timeout is unset, the function will block until such time where a message exists